### DISCUSSION TITLE
Make it easier to replace a plugin within the life cycle

### DISCUSSION THREAD

**Discussion by khmarbaise** [support-and-care/maven-support-and-care#33] [Ideas for Funding]
The life cycle is defined and many plugins for example maven-resources-plugin, maven-surefire-plugin are already bound to particular life cycle phase. 

* The idea is to have a mechanism to easily replace a plugin within the life cycle. Currently you can do that via binding a 

Examples might be: 
* use a different plugin for testing (just replace maven-surefire-plugin with a nother plugin for testing)
* Setup a kotlin build which requires to bind the kotlin-compiler-plugin (https://kotlinlang.org/docs/maven.html#compile-kotlin-and-java-sources) and also requires to remove the maven-compiler-plugin via:
```xml
<executions>
  <!-- Replacing default-compile as it is treated specially by Maven -->
  <execution>
    <id>default-compile</id>
    <phase>none</phase>
  </execution>
  <!-- Replacing default-testCompile as it is treated specially by Maven -->
  <execution>
    <id>default-testCompile</id>
    <phase>none</phase>
  </execution>
  <execution>
    <id>java-compile</id>
    <phase>compile</phase>
    <goals>
      <goal>compile</goal>
    </goals>
  </execution>
  <execution>
    <id>java-test-compile</id>
    <phase>test-compile</phase>
    <goals>
      <goal>testCompile</goal>
    </goals>
  </execution>
</executions>
```
* Maybe it would also a good idea to have the option to also change defaults like the source directories / resources directory via a plugin ? 

Need more thinking about that.

**Comment by ascheman** [comment]
This discussion has been converted to Epic #52 for tracking.

Closing this discussion.
